rust/src/ssh/ssh.rs

```diff
@@ -179,8 +214,23 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
-                                hdr.flags = SSHConnectionState::SshStateFinished;
+                            match head.msg_code { 
+                                parser::MessageCode::SshMsgNewKeys => {
+                                    hdr.flags = SSHConnectionState::SshStateFinished;
+                                }
+                                parser::MessageCode::SshMsgKexinit if HASSH_ENABLED.load(Ordering::Relaxed) => {
```

Comment:
@jasonish any opinions of using this atomic construct for settings sharing between C and Rust?

@MalakhatkoVadym I would like to hide the atomics use here in favor of a small util func similar to `rs_ssh_hassh_is_enabled`

Reply:
Its only ever modified during Suricata startup right? Can it just be a mutable module level global and then wrapped in unsafe? Saves the overhead of an atomic, and seems easy enough to audit for safety.

Reply:
I think the overhead is essentially 0 with the `Ordering::Relaxed`. Atomics get expensive when they are modified (store), and even then only if there is contention. The relaxed order means we don't impose limitations to the compiler wrt code ordering around the `load`. So I'm pretty confident this won't have a measurable perf overhead, unless Rust does something funny with atomics.

Reply:
@victorjulien Can I use `rs_ssh_hassh_is_enabled`?

Reply:
I would prefer a `safe` rust function to be called by the rust code. Then `rs_ssh_hassh_is_enabled` could also call that. I think this would keep the C/Rust boundaries clean.

Reply:
> I think the overhead is essentially 0 with the `Ordering::Relaxed`. Atomics get expensive when they are modified (store), and even then only if there is contention. The relaxed order means we don't impose limitations to the compiler wrt code ordering around the `load`. So I'm pretty confident this won't have a measurable perf overhead, unless Rust does something funny with atomics.

Thinking about this more I think I actually prefer this method, as it makes things dynamic from the start and we don't have to re-engineer that later if we want a more dynamic engine.